src/custom/pages/Claim/InvestmentFlow/InvestOption.tsx

```diff
@@ -13,78 +12,94 @@ import { ApprovalState } from 'hooks/useApproveCallback'
+  const [typedValue, setTypedValue] = useState<string>('0')
+  const [inputError, setInputError] = useState<string>('')
 
-  const { account } = useActiveWeb3React()
+  const investedAmount = investFlowData[optionIndex].investedAmount
 
   const token = currencyAmount?.currency
-
   const balance = useCurrencyBalance(account || undefined, token)
 
   const decimals = balance?.currency?.decimals
 
-  const handleStepChange = useCallback(
+  const isSelfClaiming = account === activeClaimAccount
+
+  // on invest max amount click handler
+  const setMaxAmount = useCallback(() => {
+    if (!maxCost || !balance) {
+      return
+    }
+
+    const value = maxCost.greaterThan(balance) ? balance : maxCost
+    const amount = value.quotient.toString()
+
+    updateInvestAmount({ index: optionIndex, amount })
+    setTypedValue(formatSmart(value, decimals) || '')
+    setPercentage(INVESTMENT_STEPS[INVESTMENT_STEPS.length - 1])
+  }, [balance, decimals, maxCost, optionIndex, updateInvestAmount])
+
+  // on input field change handler
+  const onInputChange = useCallback(
     (value: string) => {
-      if (!maxCost || !balance) {
+      setTypedValue(value)
+      setInputError('')
+
+      const input = parseFloat(value)
+
+      if (!value.length || isNaN(input)) {
+        updateInvestAmount({ index: optionIndex, amount: '0' })
+        setPercentage('0')
         return
       }
 
-      const scaledCurrencyAmount = _scaleValue(maxCost, value)
+      if (!value || !maxCost || !balance || !token) {
+        return
+      }
 
-      updateInvestAmount({ index: optionIndex, amount: scaledCurrencyAmount.quotient.toString() })
-      setPercentage(value)
-    },
-    [balance, maxCost, optionIndex, updateInvestAmount]
-  )
+      // parse input to BitInt string
+      const parsed = CurrencyAmount.fromRawAmount(token, parseUnits(input.toString(), decimals).toString())
 
-  const onMaxClick = useCallback(() => {
-    if (!maxCost || !balance) {
-      return
-    }
+      // calculate percent
+      const maxValue = maxCost.greaterThan(balance) ? balance : maxCost
+      const hundred = parseUnits('100', decimals).toString()
+      const percent = parsed.divide(maxValue).multiply(hundred)
 
-    const amount = maxCost.greaterThan(balance) ? balance : maxCost
+      if (percent.greaterThan(hundred)) {
+        setInputError(ErrorMsgs.Input)
+        updateInvestAmount({ index: optionIndex, amount: '0' })
+        setPercentage('0')
+        return
+      }
 
-    updateInvestAmount({ index: optionIndex, amount: amount.quotient.toString() })
-    setPercentage(INVESTMENT_STEPS[INVESTMENT_STEPS.length - 1])
-  }, [balance, maxCost, optionIndex, updateInvestAmount])
+      // update redux state with new investAmount value
+      updateInvestAmount({ index: optionIndex, amount: parsed.quotient.toString() })
+
+      // update the local state with percent value
+      setPercentage(formatSmart(percent, 2) || '0')
```

Comment:
There's a const for that :)

https://github.com/gnosis/cowswap/blob/a57a283819d358eeb69a394e05f4bba5cf0b0865/src/custom/constants/index.ts#L22